test/sendSpec.js

```diff
@@ -228,27 +276,64 @@ describe('Send', () => {
           .update(this.fulfillment)
           .digest())
       }
-    })
 
-    it('should send a transfer', function * () {
-      nock('https://example.com')
-        .post('/rpc?method=send_transfer&prefix=example.red.', [this.transfer])
-        .reply(200, true)
+      this.clpTransfer = clpPacket.serializePrepare(
+        Object.assign({}, this.transfer, {transferId: this.transfer.id}),
+        12345, // requestId
+        ilpAndCustomToProtocolData(this.transfer)
+      )
+      this.clpFulfillment = clpPacket.serializeFulfill({
+        transferId: this.transfer.id,
+        fulfillment: this.fulfillment
+      }, 98765, [])
+    })
+
+    it('should send a transfer', async function () {
+      this.mockSocket.reply(clpPacket.TYPE_PREPARE, ({requestId, data}) => {
+        assert.equal(data.transferId, this.transfer.id)
+        assert.equal(data.amount, +this.transfer.amount)
+        assert.equal(data.executionCondition, this.transfer.executionCondition)
+        assert.equal(data.expiresAt.getTime(),
+          new Date(this.transfer.expiresAt).getTime())
+        const {custom} = protocolDataToIlpAndCustom(data)
+        assert.deepEqual(custom, this.transfer.custom)
+
+        return clpPacket.serializeResponse(requestId, [])
+      })
 
       const sent = new Promise((resolve) => this.plugin.on('outgoing_prepare', resolve))
-      yield this.plugin.sendTransfer(this.transfer)
-      yield sent
-    })
-
-    it('should roll back a transfer if the RPC call fails', function * () {
-      nock('https://example.com')
-        .post('/rpc?method=send_transfer&prefix=example.red.', [this.transfer])
-        .reply(500)
-
-      yield this.plugin.sendTransfer(this.transfer)
-        .catch((e) => assert.match(e.message, /Unexpected status code 500/))
+      await this.plugin.sendTransfer(this.transfer)
+      await sent
+
+      // TODO: @sharafian, when is the balance supposed to be updated?
+      // At the moment, a call to sendTransfer does not update the balance, see below:
+      // await new Promise(async (resolve, reject) => {
+      //   setTimeout(async () => {
+      //     const balance = await this.plugin.getBalance()
+      //     console.log('balance', balance)
+      //     resolve()
+      //   }, 200)
+      // })
+    })
+
+    it.skip('should roll back a transfer if the RPC call fails', async function () {
+      this.mockSocket.reply(clpPacket.TYPE_PREPARE, ({requestId, data}) => {
+        return clpPacket.serializeError({rejectionReason: this.ilpError},
+          requestId, [])
+      })
 
-      assert.equal((yield this.plugin.getBalance()), '0', 'balance should be rolled back')
+      try {
+        // TODO: @sharafian, this test case seems to not have worked as intended,
+        // because sendTransfer did not throw. Any reason why sendTransfer only
+        // throws if the plugin is stateful?
+        await this.plugin.sendTransfer(this.transfer)
```

Comment:
This doesn't distinguish between throwing and rejecting, which is a nice feature of async functions when you want to handle both error mechanisms in the same way, but it does not test whether LPI was implemented correctly (i.e. rejecting), or 'close, but no cigar' (i.e. throwing the error that it should have rejected with)